src/table.js

```diff
@@ -1386,6 +1386,94 @@ Table.prototype.truncate = function(gaxOptions, callback) {
   );
 };
 
+/**
+ * Generates Consistency-Token and check consistency for generated token
+ * In-case consistency check returns false, retrial is done in interval
+ * of 5 seconds till 10 minutes, after that it returns false.
+ *
+ * @param {function(?error, ?boolean)} callback The callback function.
+ * @param {?Error} callback.err An error returned while making this request.
+ * @param {?Boolean} callback.resp Boolean value.
+ */
+Table.prototype.waitForReplication = function(callback) {
+  // handler for generated consistency-token
+  const tokenHandler = (err, resp) => {
+    if (err) {
+      return callback(err);
+    }
+
+    // set timeout for 10 minutes
+    const timeoutAfterTenMinutes = setTimeout(() => {
+      callback(null, false);
+    }, 10 * 60 * 1000);
+
+    // method checks if retrial is required & init retrial with 5 sec delay
+    const retryIfNecessary = (err, res) => {
+      if (err || res.consistent === true) {
```

Comment:
This is wrong since if the error is set the next argument should be null or undefined. Change this block to something like this:

```js
if (err || res.consistent === true) {
  clearTimeout(timeoutAfterTenMinutes);
  if (err) {
    return callback(err);
  } else {
    return callback(null, true)
  }
}
```

Anything more is too clever IMO